Atomic/syscontainers.py

```diff
@@ -1369,6 +1370,53 @@ def uninstall(self, name):
         if rpm_installed:
             RPMHostInstall.uninstall_rpm(rpm_installed.replace(".rpm", ""))
 
+    # get the corresponding layers from an image
+    def get_layers_from_image(self,img,repo):
+        try:
+            _, rev = self._resolve_image(repo, img,allow_multiple=True)[0]
+        except (IndexError, TypeError):
+            raise ValueError("Image {} not found".format(img))
+        manifest = self._image_manifest(repo, rev)
+        if not manifest:
+            return None
+        layers = SystemContainers.get_layers_from_manifest(json.loads(manifest))
+        return layers
+
+    # define a function to handle layer image checking
+    def get_dangling_system_images(self):
```

Comment:
Oh, I considered that as well, but that time I was not sure about which is better, whether resolve_image, then ask for manifest, or resolve rev then manifest. And yea, I have noticed the speed of running atomic images list become slower than before, I am currently "working" on it to sort of making it have speed as usual.